lookup.go

```diff
@@ -99,21 +111,25 @@ func doLookup(g *GlobalLookupFactory, gc *GlobalConf, input <-chan interface{},
 		} else {
 			line := genericInput.(string)
 			var changed bool
-			var rawName string
+			var lookupName string
+			rawName := ""
+			nameServer := ""
 			var rank int
 			if gc.AlexaFormat == true {
 				rawName, rank = parseAlexa(line)
 				res.AlexaRank = rank
+			} else if gc.NameServerMode {
+				nameServer = AddDefaultPortToDNSServerName(line)
 			} else {
-				rawName = line
+				rawName, nameServer = parseNormalInputLine(line)
 			}
-			lookupName, changed := makeName(rawName, gc.NamePrefix)
+			lookupName, changed = makeName(rawName, gc.NamePrefix, gc.NameOverride)
```

Comment:
The interaction between `makeName` and `NameServerMode` is non-obvious, but short of splitting up each style of parsing into it's own function and doing one or the other, I'm not sure this gets any simpler.